### PR TITLE
Revert "Move `glob_to_regex` and `re_word_boundary` to `matrix-python-common` (#11505)

### PR DESCRIPTION
**Pull request by squahtx**
This reverts commit a77c36989785c0d5565ab9a1169f4f88e512ce8a.

Our downstream package maintainers need more notice before adding a new package.
To be revisited in the new year.

### CODE REVIEW

File: changelog.d/11527.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Temporarily revert usage of `matrix-python-common`.
```
Comment (clokep) [inline changelog.d/11527.misc:1]: Can we nuke both changelogs after this merges so we don't add anything to the changes doc?

**Review by DMRobertson — APPROVED** [review]
Seems reasonable to me, let's sanity check that CI succeeds though?

**Review by clokep — APPROVED** [review]
Seems reasonable, I assume this was a `git revert`.

File: changelog.d/11527.misc (hunk 1 above)
Comment (squahtx) [reply changelog.d/11527.misc:1]: Yep, I'll be cutting them out of the changes doc after.

**Pull request merged**